packages/influx-utilities/src/lib/Client.ts

```diff
@@ -1,40 +1,34 @@
-import { InfluxDB, Point, QueryApi, WriteApi, WritePrecisionType, type ClientOptions } from '@influxdata/influxdb-client';
-import { tryNumberParse } from './common';
+import { InfluxDB, type ClientOptions, type Point, type QueryApi, type WriteApi, type WritePrecisionType } from '@influxdata/influxdb-client';
+import { tryNumberParse } from './common.js';
 
 export class Client {
 	public readonly influx: InfluxDB;
-	public readonly queryApi: QueryApi | null;
-	public readonly writeApi: WriteApi | null;
+	public readonly queryApi: QueryApi;
+	public readonly writeApi: WriteApi;
 
 	private readonly injectedTags: [name: string, value: string][] = [];
 
 	public constructor(options: Client.Options);
-	public constructor({ org, writeBucket, writePrecision, ...options }: Client.Options = {}) {
-		this.influx = new InfluxDB(
-			process.env.INFLUX_OPTIONS_STRING ?? {
-				...options,
-				url: options.url ?? process.env.INFLUX_URL!,
-				proxyUrl: options.proxyUrl ?? process.env.INFLUX_PROXY_URL,
-				timeout: options.timeout ?? tryNumberParse(process.env.INFLUX_TIMEOUT),
-				token: options.token ?? process.env.INFLUX_TOKEN
-			}
-		);
-
-		const _org = org ?? process.env.INFLUX_ORG;
-		if (_org) {
-			this.queryApi = this.influx.getQueryApi(_org);
-
-			const _bucket = writeBucket ?? process.env.INFLUX_WRITE_BUCKET;
-			if (_bucket) {
-				const _precision = writePrecision ?? (process.env.INFLUX_WRITE_PRECISION as WritePrecisionType | undefined);
-				this.writeApi = this.influx.getWriteApi(_org, _bucket, _precision);
-			} else {
-				this.writeApi = null;
-			}
-		} else {
-			this.queryApi = null;
-			this.writeApi = null;
-		}
+	public constructor({
+		url = process.env.INFLUX_URL,
+		token = process.env.INFLUX_TOKEN,
+		org = process.env.INFLUX_ORG,
+		writeBucket = process.env.INFLUX_WRITE_BUCKET,
+		writePrecision = process.env.INFLUX_WRITE_PRECISION,
+		proxyUrl = process.env.INFLUX_PROXY_URL,
+		timeout = tryNumberParse(process.env.INFLUX_TIMEOUT),
+		...options
+	}: Client.Options = {}) {
+		if (!url) throw new TypeError('No Influx URL was provided');
+		if (!token) throw new TypeError('No Influx Token was provided');
+
+		this.influx = new InfluxDB({ ...options, url, token, proxyUrl, timeout });
+
+		if (!org) throw new TypeError('No Influx Org was provided');
+		if (!writeBucket) throw new TypeError('No Influx Write Bucket was provided');
```

Comment:
Swap the order, we don't need to create an `InfluxDB` instance if we may throw an error afterwards.